src/enclave/enclave.h

```diff
@@ -245,7 +246,28 @@ namespace enclave
         {
           node.start_ledger_recovery();
         }
-        bp.run(circuit->read_from_outside());
+
+        bp.run(circuit->read_from_outside(), [](size_t consecutive_idles) {
+          static std::chrono::microseconds idling_start_time;
+          const auto time_now = enclave::get_enclave_time();
+
+          if (consecutive_idles == 0)
+          {
+            idling_start_time = time_now;
+          }
+
+          // Handle initial idles by pausing, eventually sleep (in host)
+          constexpr std::chrono::milliseconds timeout(5);
```

Comment:
do we want this to be configurable?

Reply:
In future yes, but we also want the sleep to do something smarter - I'll plumb in configurable parameters in a future PR.